include/storage/io_config.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef OSRM_IO_CONFIG_HPP
+#define OSRM_IO_CONFIG_HPP
+
+#include "util/exception.hpp"
+#include "util/log.hpp"
+
+#include <array>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/path.hpp>
+#include <string>
+
+namespace osrm
+{
+namespace storage
+{
+struct IOConfig
+{
+    IOConfig(std::vector<boost::filesystem::path> _required_input_files,
+             std::vector<boost::filesystem::path> _optional_input_files,
+             std::vector<boost::filesystem::path> _output_files)
```

Comment:
Hmm I _think_ `_lowercase` was reserved, that's why the common idiom is to make it `lowercase_` - not entirely sure though